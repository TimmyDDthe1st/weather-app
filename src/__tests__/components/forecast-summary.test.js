import React from 'react';
import { render } from '@testing-library/react';
import ForecastSummary from '../../components/forecast-summary';

describe('Forecast Summary', () => {
    it("renders correctly", () => {
        const { asFragment } = render(
          <ForecastSummary
            date='mockDate'
            temperature="mockTemperature"
            description="mockDescription"
            icon="mockIcon"
          />
        );
    
        expect(asFragment).toMatchSnapshot();
      });

it("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date="mockDate"
        temperatureMax="mockMaxTemperature"
        temperatureMin="mockMinTemperature"
        description="mockDescription"
        icon="mockIcon"
      />
    );

    expect(getByText("mockDate")).toHaveClass("date");
    expect(getByText("mockMaxTemperature")).toHaveClass("temperatureMax");
    expect(getByText("mockMinTemperature")).toHaveClass("temperatureMin");
    expect(getByText("mockDescription")).toHaveClass("description");
    expect(getByText("mockIcon")).toHaveClass("icon");
  });
})