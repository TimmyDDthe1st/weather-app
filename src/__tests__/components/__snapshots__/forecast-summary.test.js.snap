// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summary renders correctly 1`] = `[Function]`;
